content/5_Plat/Sweep_Line.mdx

```diff
@@ -162,8 +162,160 @@ For each point, we recalculate $\delta = \min(\delta, \delta_p)$, and update our
 <FocusProblem problem="segSam" />
 
 ### Solution
+Let's simplify the problem a little bit and focus on finding any overlapping segments. The segment we are looking for is one of the two.
 
-<IncompleteSection />
+To find a pair of overlapping segments we use a sweep line approach. Imagine sweeping a vertical line across the scene from left to right, pausing at every segment endpoint.
+We simulate this by sorting all the segment endpoints by x and walking through this sorted array (called 'events' in the code below). As we scan, we keep track of active segments
+by using a set (called 'active_segments' in the code below). When we hit the beginning point of a segment, we add it to the active set, and when we hit the ending point of a segment, we remove it from the active set.
+The operation of inserting or removing takes $\mathcal{O}(\log n)$.
+
+The active set of segments is ordered by $y$ coordinate. If two segments overlap, it means that they are adjacent in the set. So every time we insert or remove
+a segment we check if the adjacent segments overlap.
+
+Here is an animation of how it works:
+
+<video width="960" height="720" controls>
+	<source src="/animations/cowsteeplechase.mp4" type="video/mp4"/>
+</video>
+
+
+## Implementation
+
+Time complexity: $\mathcal{O}(n\log n)$.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+long long sweep_line_x;
+
+struct Point {
+	long long x, y, segment_idx;
+	bool operator<(const Point &other) {
+		return x == other.x ? y < other.y : x < other.x;
+	}
+};
+struct Segment {
+	long long a, b, x, y, idx;
+};
+
+double coordinate(Segment a) {
+	if (a.a == a.x) { return a.b; }
+	return a.b + (a.y - a.b) * (sweep_line_x - a.a) / (a.x - a.a);
+}
+
+int sign(long long x) {
+	if (x == 0) {
+		return 0;
+	} else {
+		return x < 0 ? -1 : +1;
+	}
+}
+
+int operator*(Point p1, Point p2) { return sign(p1.x * p2.y - p1.y * p2.x); }
```

Comment:
this is inconsistent with the next line don't put everything in one line

Reply:
Alright  I moved them all together

Reply:
wait why are the functions not included in the codesnip

Reply:
Ok I moved all the function in the codesnips